broadinstitute/single_cell_portal_core
# Adding `FeatureAnnouncements` to home page, feed (SCP-3729)

This update adds the `FeatureAnnouncement` model & associated functionality.  These are objects with titles/html content blocks that serve as small, in-application announcements for new features, functionality, or other items of interest.  Each announcement can embed text & images, as well as link out to documentation or other resources.  A button & associate view has been added as a "latest features" page, that will contain all announcements.   This shows a truncated view of each announcement (150 words), and each individual announcement can be clicked on to show the entire content.  There is also an entry in the "Help & Resources" menu to view this page, if any announcements are published.

Also, an announcement be "archived" to remove it from the "latest features" and place in the archive at the bottom of the page.  Announcements can also be unpublished without being deleted.  Additionally, each announcement will keep track of the latest date when either state changed (published/archived), as well as full history tracking of both fields via the `track_history` functionality (requires database queries to retrieve).  There is also a minimal admin CMS for managing all announcements, accessible from the "Feature Announcements" menu item for admin users.

Only features in the "latest" feed will be advertised from the home page, with a count of the number of announcements.  If no announcements are present, or all are archived, then the button will not render.

Home page w/ new button:
![Screen Shot 2021-10-06 at 3 58 38 PM](https://user-images.githubusercontent.com/729968/136275038-3a382a15-bb20-4b7d-bc29-3511c57d474c.png)

Latest Features view:
![Screen Shot 2021-10-06 at 4 05 05 PM](https://user-images.githubusercontent.com/729968/136275210-688be1af-a884-4060-9fd2-e270a98b6fc2.png)

MANUAL TESTING
1. Log in with an admin account
2. Select "Feature Announcements"  from the profile menu
3. Click "New Announcement" at the bottom, and create an entry (leave published & archived unchanged)
4. Once saved, return to the home page and confirm the button renders and takes you to the appropriate view
5. Back in the admin CMS, add a new announcement again, and set archived to true
6. Back in the latest features page, confirm the new entry is at the bottom under "archives"
7. Back in the admin CMS, archive the first announcement
8. Back on the home page, confirm the "New Features" button no longer appears, but help menu link still takes you to the correct page

This PR satisfies SCP-3729
